apiservers/engine/backends/container.go

```diff
@@ -111,13 +115,55 @@ func (c *Container) ContainerCreate(config types.ContainerCreateConfig) (types.C
 				http.StatusInternalServerError)
 	}
 
-	// Check if the image exist
-	layerID, found := hackMap[config.Config.Image]
+	c.m.Lock()
+	defer c.m.Unlock()
+
+	layer, found := hackMap[config.Config.Image]
 	if !found {
-		return types.ContainerCreateResponse{},
-			derr.NewErrorWithStatusCode(fmt.Errorf("Container look up failed"),
-				http.StatusInternalServerError)
+		// FIXME: This is a temporary workaround until we have a name resolution story.
+		// Call imagec with -resolv parameter to learn the name of the vmdk and put it into in-memory map
+		cmdArgs := []string{"-reference", config.Config.Image, "-resolv", "-standalone"}
+
+		out, err := goexec.Command("/sbin/imagec", cmdArgs...).Output()
+		if err != nil {
+			log.Printf("imagec exit code:", err)
+			return types.ContainerCreateResponse{},
+				derr.NewErrorWithStatusCode(fmt.Errorf("Container look up failed"),
+					http.StatusInternalServerError)
+		}
+		var v1 struct {
+			ID string `json:"id"`
+			// https://github.com/docker/engine-api/blob/master/types/container/config.go
+			Config container.Config `json:"config"`
+		}
+		if err := json.Unmarshal(out, &v1); err != nil {
+			return types.ContainerCreateResponse{},
+				derr.NewErrorWithStatusCode(fmt.Errorf("Failed to unmarshall image history: %s", err),
+					http.StatusInternalServerError)
+		}
+		log.Printf("v1 = %+v", v1)
+
+		hackMap[config.Config.Image] = V1Compatibility{
+			v1.ID,
+			v1.Config,
+		}
+
+		layer = hackMap[config.Config.Image]
+	}
+
+	// Overwrite the config struct
+	if len(config.Config.Cmd) == 0 {
+		config.Config.Cmd = layer.Config.Cmd
+	}
+	if config.Config.WorkingDir == "" {
+		config.Config.WorkingDir = layer.Config.WorkingDir
 	}
+	if len(config.Config.Entrypoint) == 0 {
+		config.Config.Entrypoint = layer.Config.Entrypoint
+	}
+	config.Config.Env = append(config.Config.Env, layer.Config.Env...)
```

Comment:
The real question is how do this function if VIC was in the layer config - this should either be a map or we need to apply it last wins style in the tether.